src/WebApiToTypeScript/Interfaces/InterfaceService.cs

```diff
@@ -163,7 +163,7 @@ private void WriteInterfaces(TypeScriptBlock interfacesBlock, InterfaceNode inte
                 WriteInterfaceNode(interfacesBlock, interfaceNode);
             }
 
-            foreach (var derivedInterfaceNode in interfaceNode.DerivedInterfaces)
+            foreach (var derivedInterfaceNode in interfaceNode.DerivedInterfaces.ToList())
```

Comment:
The discovery and generation aspects is one of the top things we need to refactor; I will bank on you guys testing this on your projects for this change.